p2p/peermanager.go

```diff
@@ -1212,6 +1252,16 @@ func (p *peerInfo) Score() PeerScore {
 	if p.Persistent {
 		score += PeerScorePersistent
 	}
+
+	switch {
+	case p.MutableScore+int64(score) > math.MaxUint8:
+		score = math.MaxUint8
+	case p.MutableScore+int64(score) < 0:
+		score = 0
+	default:
+		score += PeerScore(p.MutableScore)
```

Comment:
I think we want to keep the scores between [0,100].

Reply:
Why 100?

This logic keeps it between 0 and 255 (as a uint8), which seems somewhat arbitrary as well.

Reply:
Because it cannot be higher than a persistent peer which is 100.

Reply:
That being said, we can have custom logic to let the value go beyond 100 and prefer persistent peers, but I think the ranking logic remains simpler if we just cap it at a max, i.e. persistent peers -- 100.

Reply:
why not set persistent peer as "max Uint8?"

Reply:
We can, I don't really think the type matters as much.

Reply:
I made the persistent peer's be the max score, but it seems like we'd also want to rank the persistent peers among themselves, eventually, but that's easy enough to change later